src/test_rand_types.rs

```diff
@@ -0,0 +1,324 @@
+            vout: rng.gen(),
+        }
+    }
+}
+
+impl Distribution<Script> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Script {
+        Script::from(gen_vec(rng))
+    }
+}
+
+impl Distribution<Transaction> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Transaction {
+        Transaction {
+            version: rng.gen(),
+            lock_time: rng.gen(),
+            input: gen_vec(rng),
+            output: gen_vec(rng),
+        }
+    }
+}
+
+impl Distribution<BlockHeader> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> BlockHeader {
+        BlockHeader {
+            version: rng.gen(),
+            prev_blockhash: gen_hash(rng),
+            merkle_root: gen_hash(rng),
+            time: rng.gen(),
+            bits: rng.gen(),
+            nonce: rng.gen(),
+        }
+    }
+}
+
+impl Distribution<Block> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Block {
+        Block {
+            header: rng.gen(),
+            txdata: gen_vec(rng),
+        }
+    }
+}
+
+impl Distribution<GetCFilters> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> GetCFilters {
+        GetCFilters {
+            filter_type: rng.gen(),
+            start_height: rng.gen(),
+            stop_hash: gen_hash(rng),
+        }
+    }
+}
+
+impl Distribution<CFilter> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> CFilter {
+        CFilter {
+            filter_type: rng.gen(),
+            block_hash: gen_hash(rng),
+            filter: gen_vec(rng),
+        }
+    }
+}
+
+impl Distribution<GetCFHeaders> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> GetCFHeaders {
+        GetCFHeaders {
+            filter_type: rng.gen(),
+            start_height: rng.gen(),
+            stop_hash: gen_hash(rng),
+        }
+    }
+}
+
+impl Distribution<CFHeaders> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> CFHeaders {
+        CFHeaders {
+            filter_type: rng.gen(),
+            stop_hash: gen_hash(rng),
+            previous_filter: gen_hash(rng),
+            filter_hashes: gen_vec_hash(rng),
+        }
+    }
+}
+
+impl Distribution<GetCFCheckpt> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> GetCFCheckpt {
+        GetCFCheckpt {
+            filter_type: rng.gen(),
+            stop_hash: gen_hash(rng),
+        }
+    }
+}
+
+impl Distribution<CFCheckpt> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> CFCheckpt {
+        CFCheckpt {
+            filter_type: rng.gen(),
+            stop_hash: gen_hash(rng),
+            filter_headers: gen_vec_hash(rng),
+        }
+    }
+}
+
+impl Distribution<CommandString> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> CommandString {
+        NetworkMessage::possibilites().choose(rng).unwrap().clone() // Can never fail. the array isn't empty.
```

Comment:
Unknown command strings are legal, right?

Reply:
It will fail decoding https://github.com/rust-bitcoin/rust-bitcoin/blob/master/src/network/message.rs#L324